packages/cloud-functions/src/cico/simplexEventPolling.ts

```diff
@@ -0,0 +1,99 @@
+import * as functions from 'firebase-functions'
+import { trackEvent } from '../bigQuery'
+import { SIMPLEX_DATA } from '../config'
+import { fetchWithTimeout, flattenObject, lookupAddressFromTxId } from './utils'
+
+const SIMPLEX_BIG_QUERY_EVENT_TABLE = 'cico_provider_events_simplex'
+
+// https://integrations.simplex.com/wallet-api-integration#events-api_group
+interface SimplexEventPayload {
+  events: SimplexTransactionEvent[]
+}
+
+interface SimplexTransactionEvent {
+  event_id: string
+  name: SimplexTxStatus
+  payment: {
+    id: string
+    status: string
+    created_at: string
+    partner_id: number
+    updated_at: string
+    crypto_currency: string
+    fiat_total_amount: {
+      amount: number
+      currency: string
+    }
+    crypto_total_amount: {
+      amount: number
+      currency: string
+    }
+    partner_end_user_id: string
+  }
+  timestamp: string
+}
+
+enum SimplexTxStatus {
+  Started = 'payment_request_submitted',
+  Completed = 'payment_simplexcc_approved',
+  Failed = 'payment_simplexcc_declined',
+}
+
+const getSimplexEvents = async () => {
+  const response = await fetchWithTimeout(SIMPLEX_DATA.event_url, {
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `ApiKey ${SIMPLEX_DATA.api_key}`,
+    },
+  })
+
+  if (!response.ok) {
+    throw new Error(JSON.stringify(response))
+  }
+
+  const simplexEvents: SimplexEventPayload = await response.json()
+  return simplexEvents
+}
+
+const deleteSimplexEvent = async (event: SimplexTransactionEvent) => {
+  try {
+    await fetchWithTimeout(`${SIMPLEX_DATA.event_url}/${event.event_id}`, {
+      method: 'DELETE',
+      headers: {
+        Authorization: `ApiKey ${SIMPLEX_DATA.api_key}`,
+      },
+    })
+  } catch (error) {
+    console.error('Delete error: ', JSON.stringify(error))
+    throw error
+  }
+}
+
+export const simplexEventPolling = functions.https.onRequest(async (_, res) => {
```

Comment:
Who will call this?
From what I see there's a risk to have this called multiple times, causing `trackEvent` to be called more than expected.
Can we lock this while it's processing?

Reply:
It's a cloud function that runs on an hourly cron job. There is indeed a small risk that it is manually run twice in quick succession.

I don't think attempting to lock cloud functions is a good practice but another idea is to finish each function call with a clean-up function that deletes any duplicate database entries. What do you think about that solution?

Reply:
I see, problem is anyone could trigger it and cause these troubles. So we need a good enough way to ensure it won't happen. What you're proposing seems would work, but is it easy to implement and reliable?

In all cases, I think we should verify the invocation is coming from the cloud scheduler, one way would be with a shared secret:
https://stackoverflow.com/questions/53181297/verify-http-request-from-google-cloud-scheduler

Now there's also an "easy" way to ensure only 1 instance of a cloud function runs in parallel.
This can be done with something like `export const myFunction = functions.runWith({ maxInstances: 1 }).…`
https://firebase.google.com/docs/reference/functions/function_configuration_.runtimeoptions#maxinstances

With this, we could avoid a clean-up function.

Reply:
Gotcha, looks like the easiest solution is using the `maxInstances` param. Will go with that and can revisit further invocation restrictions if we see any issues with duplicate data.

Thanks!